src/Microsoft.AspNet.OData.Shared/Query/Expressions/ExpressionBinderBase.cs

```diff
@@ -1077,6 +1078,20 @@ internal static bool IsDate(Type type)
             return IsType<Date>(type);
         }
 
+        internal static bool IsDate(BinaryOperatorNode binaryOperatorNode)
```

Comment:
>IsDate [](start = 29, length = 6)

the method name is confusing with others. Maybe it's better to rename it.

Reply:
I renamed method to ContainsDateFunction